### PR TITLE
[develop] Diego - Create team test

### PR DESCRIPTION
**Pull request by nanrehd**
Team pages structure implemented.

### CODE REVIEW

File: src/test/resources/features/teams/Create_Team.feature [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    Given I log in with my Trello account as "user1"
+
+  @acceptance
+  Scenario: User is able to create a team
+    When I create a "team" from header with the following data
+      | name | Sith Empire |
+      | type | Education   |
+    And I skip inviting members
+    Then I should have a team created with the following data
+      | name | Sith Empire |
```
Comment (carlosmc23) [inline src/test/resources/features/teams/Create_Team.feature]: You need to add a white space here.

File: src/test/java/org/fundacionjala/trello/stepdefs/TeamStepDefs.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        teamPage = new TeamPage(getChromeDriver());
+        assertEquals(teamPage.getTeamName(), expectedData.get("name"));
+    }
+
+    /**
+     * Skips the step of inviting members.
+     */
+    @When("I skip inviting members")
+    public void iSkipInvitingMembers() {
+        TeamInviteForm teamInviteForm = new TeamInviteForm(getChromeDriver());
```
Comment (carlosmc23) [inline src/test/java/org/fundacionjala/trello/stepdefs/TeamStepDefs.java]: I think you should instaciate a TreamInviteForm in the constructor of the class

File: src/test/java/org/fundacionjala/trello/stepdefs/TeamStepDefs.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    /**
+     * Validates the creation of the team with the given data.
+     *
+     * @param expectedData expected data to validate the creation.
+     */
+    @Then("I should have a team created with the following data")
+    public void validateCreationWithData(final Map<String, String> expectedData) {
+        teamPage = new TeamPage(getChromeDriver());
```
Comment (carlosmc23) [inline src/test/java/org/fundacionjala/trello/stepdefs/TeamStepDefs.java]: I think you should instaciate a TeamPage in the constructor of the class

File: src/test/java/org/fundacionjala/trello/TDD/team/CreateTeamTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        driver.quit();
+    }
+
+    @Test
+    public void createTeamTest(final Map<String, String> data) {
+        User user = new User("user1");
+        LoginPage page = new LoginPage(getChromeDriver());
+        PageObject actualPage = page.setCredentials(user.getEmail(), user.getPassword())
+                .submit();
+//        context.saveActualPage(actualPage);
```
Comment (carlosmc23) [inline src/test/java/org/fundacionjala/trello/TDD/team/CreateTeamTest.java]: this comment is unnecessary

**Review by carlosmc23 — COMMENTED** [review]
good joob, I have some suggestions

File: src/main/java/org/fundacionjala/trello/pages/home/TeamsPage.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+public final class TeamsPage extends HomePage {
+
+    public TeamsPage(final WebDriver driver) {
+        super(driver);
+    }
+
+    public WebElement getSettingsButton(final String teamName) {
+        return driver.findElement(By.xpath(String.format("//h3[text()='%s']/parent::div/parent::div", teamName)));
+    }
```
Comment (luqp) [inline src/main/java/org/fundacionjala/trello/pages/home/TeamsPage.java:17]: You can convert this xpath locator in a constant.

File: src/main/java/org/fundacionjala/trello/pages/team/TeamInviteForm.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import org.fundacionjala.trello.pages.forms.IFillerField;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.Map;
+
+public final class TeamInviteForm extends FormPage<TeamPage> {
+
+
```
Comment (luqp) [inline src/main/java/org/fundacionjala/trello/pages/team/TeamInviteForm.java]: There is an unnecessary whitespace line.

**Review by carlosmc23 — APPROVED** [review]

**Review by luqp — COMMENTED** [review]
Good job, I leave some comments, please take a look.

File: src/main/java/org/fundacionjala/trello/pages/home/TeamsPage.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import org.openqa.selenium.WebElement;
+
+public final class TeamsPage extends HomePage {
+
+    public TeamsPage(final WebDriver driver) {
+        super(driver);
+    }
+
+    public WebElement getSettingsButton(final String teamName) {
+        return driver.findElement(By.xpath(String.format("//h3[text()='%s']/parent::div/parent::div", teamName)));
```
Comment (Snip-77) [inline src/main/java/org/fundacionjala/trello/pages/home/TeamsPage.java]: Can this be declared on a constant?

File: src/main/java/org/fundacionjala/trello/pages/team/TeamSettings.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * Deletes a team permanently.
+     *
+     */
+    public void deleteTeam() {
+        action.click(deleteTeamButton);
+        action.click(confirmDeleteButton);
+        new TeamsPage(driver);
+    }
+
```
Comment (Snip-77) [inline src/main/java/org/fundacionjala/trello/pages/team/TeamSettings.java]: Unnecessary space here xD

**Review by Snip-77 — COMMENTED** [review]
Have some minor comments. Looks good to go!

**Review by Snip-77 — APPROVED** [review]
